From d123105d6cead7ef612a33e46144be74af004200 Mon Sep 17 00:00:00 2001
From: HungTran <hung.tran.jy@rvc.renesas.com>
Date: Fri, 6 May 2016 02:23:04 +0700
Subject: [PATCH] iwg20m: Fix issue HDMI output is clone from LVDS

HDMI output is clone from LVDS instead of multi-display
(In other word, the HDMI output shows same screen as LVDS).

But the Wayland system detects HDMI connect and assumes
multi-display is used, and reserve a 1920x1080 for HDMI.
User cannot see this area as HDMI does not display it.
Even worse, GUI applications will show up in this area
as HDMI is default output, thus user cannot see and use.

Signed-off-by: HungTran <hung.tran.jy@rvc.renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_crtc.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
index cf6da52b..f17a84f 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
@@ -211,15 +211,15 @@ void rcar_du_crtc_route_output(struct drm_crtc *crtc,
 	 */
 	rcrtc->outputs |= BIT(output);
 
-#ifdef CONFIG_MACH_IWG20M
-	/* Store LVDS routing to LVDS0 for IWG20M. */
-	if (rcar_du_has(rcdu, RCAR_DU_FEATURE_DEFR8) &&
-	    output == RCAR_DU_OUTPUT_LVDS0)
-#else
+//#ifdef CONFIG_MACH_IWG20M
+	///* Store LVDS routing to LVDS0 for IWG20M. */
+	//if (rcar_du_has(rcdu, RCAR_DU_FEATURE_DEFR8) &&
+	    //output == RCAR_DU_OUTPUT_LVDS0)
+//#else
 	/* Store RGB routing to DPAD0 for R8A7790. */
 	if (rcar_du_has(rcdu, RCAR_DU_FEATURE_DEFR8) &&
 	    output == RCAR_DU_OUTPUT_DPAD0)
-#endif
+//#endif
 		rcdu->dpad0_source = rcrtc->index;
 }
 
-- 
1.9.1

